src/parse-nm.c

```diff
@@ -1014,9 +1020,16 @@ netplan_parser_load_keyfile(NetplanParser* npp, const char* filename, GError** e
             default: break;
         }
 
-        keyfile_handle_generic_str(kf, "wifi-security", "psk", &ap->auth.password);
-        if (ap->auth.password)
-            ap->has_auth = TRUE;
+        if (ap->auth.key_management == NETPLAN_AUTH_KEY_MANAGEMENT_WPA_PSK
+            || ap->auth.key_management == NETPLAN_AUTH_KEY_MANAGEMENT_WPA_SAE) {
+            keyfile_handle_generic_str(kf, "wifi-security", "psk", &ap->auth.password);
+            if (ap->auth.password)
+                ap->has_auth = TRUE;
+        } else {
+            keyfile_handle_generic_str(kf, "wifi-security", "psk", &ap->auth.psk);
```

Comment:
I did that to help differentiate when the PSK will go to "ap".password or "ap".auth.password so it wouldn't change behavior (and tests would not break :P). I'll see how I can always add the PSK to the new psk variable.

Reply:
yeah it cause a small conflict with our previous idea of always set "pmf" to optional if "ap".password is used. By always reading the PSK into `ap->auth.psk` we'll end up always emitting "ap".password in the YAML and enabling the pmf setting even if it wasn't in the original keyfile we are loading, which is not a big deal I guess. That was the reason why I tried to differentiate the cases...

Reply:
I pushed this change in a new commit

Reply:
> I did that to help differentiate when the PSK will go to "ap".password or "ap".auth.password so it wouldn't change behavior (and tests would not break :P). I'll see how I can always add the PSK to the new psk variable.

Understood. And I think that makes sense! Enabling the `pmf` setting even if it wasn't in the original keyfile might actually be an issue, as it might break NetworkManager unit-tests at build time, when it's trying to re-generate the correct keyfile. I'm not sure if it has tests around that PMF setting, but it could have in the future.

So we should really try to not change behavior here. Do you think we could have a `has_auth_psk_shortcut` helper variable, to indicate which case we're in? Then making use of this in the YAML emitter (`netplan.c`), to write the correct password field, instead of changing the tests. And also no change keyfile behavior.